lib/actuator/Motor.cpp

```diff
@@ -0,0 +1,111 @@
+// Copyright (C) 2024 COGIP Robotics association <cogip35@gmail.com>
+// This file is subject to the terms and conditions of the GNU Lesser
+// General Public License v2.1. See the file LICENSE in the top level
+// directory for more details.
+
+#include "board.h"
+#include "actuator/Motor.hpp"
+#include "actuator/PositionalActuator.hpp"
+
+#include <iostream>
+
+namespace cogip {
+namespace actuators {
+namespace positional_actuators {
+
+/// @brief Construct a Motor by unpacking the static parameters and starting the control thread.
+/// @param motor_parameters  Reference to a `MotorParameters` in static storage.
+Motor::Motor(const MotorParameters& motor_parameters)
+    : PositionalActuator(
+          motor_parameters.id,
+          motor_parameters.default_timeout_ms,
+          motor_parameters.send_state_cb
+      ),
+      params_(motor_parameters),
+      distance_controller_(motor_parameters.pose_controller_parameters),
+      speed_controller_(motor_parameters.speed_controller_parameters),
+      motor_distance_filter_(motor_parameters.motor_pose_filter_parameters),
+      speed_filter_(motor_parameters.speed_filter_parameters),
+      motor_engine_(motor_parameters.motor, motor_parameters.odometer, motor_parameters.engine_thread_period_ms)
+{
+    // Build the cascade: MotorPoseFilter → PosePID → SpeedFilter → SpeedPID
+    dualpid_meta_controller_.add_controller(&motor_distance_filter_);
+    dualpid_meta_controller_.add_controller(&distance_controller_);
+    dualpid_meta_controller_.add_controller(&speed_filter_);
+    dualpid_meta_controller_.add_controller(&speed_controller_);
```

Comment:
Interesting way to use the engines. We can think about use this in the same way for the Motion control platform ?

Reply:
The `add_controler` method ? We are already using it:
https://github.com/cogip/mcu-firmware/blob/c3b80e1ecb8c940d0485bf058b187f6ad056db3a/platforms/pf-robot-motion-control/motion_control.cpp#L221

Reply:
Yes but it's more about wrapping it inside a class